JamGame/Assets/Scripts/TileUnion/Inspector.cs

```diff
@@ -70,12 +72,55 @@ private void DrawGizmoRecursively(Transform transform)
         [Button("Update tiles position and rotation", Style = ButtonStyle.Box)]
         public void UpdateTilesPositionAndRotation()
         {
-            foreach (TileImpl tile in Tiles)
+            foreach (TileImpl tile in tiles)
             {
                 tile.SetPosition(gridProperties, tile.Position);
                 tile.SetRotation(tile.Rotation);
             }
         }
+
+        [Button(Style = ButtonStyle.Box)]
+        public void ValidateProjectedTiles(int projectedCount)
+        {
+            if (IsAllWithMark("Outside"))
+            {
+                foreach (TileImpl tile in tiles)
+                {
+                    if (tile.ProjectedTiles.Count() != projectedCount && tile.TileToProject != null)
+                    {
+                        ProjectTile(tile, projectedCount);
+                    }
+                }
+            }
+        }
+
+        [Button(Style = ButtonStyle.Box)]
+        public void RecreateProjectedTiles(int projectedCount)
+        {
+            if (!IsAllWithMark("Outside"))
+            {
+                Debug.LogError("Projected Tiles valid only for outside tiles");
+                return;
+            }
+            foreach (TileImpl tile in tiles)
+            {
+                if (tile.TileToProject == null)
+                {
+                    Debug.LogError($"Tile {tile.gameObject.name} have null TileToProject");
+                    continue;
+                }
+                ProjectTile(tile, projectedCount);
+            }
+        }
```

Comment:
Please perform this for ALL the assets and automatically when assets are reloaded

Reply:
Discussed in dm